smibhid/lib/config/config_management.py

```diff
@@ -0,0 +1,54 @@
+from ulogging import uLogger
+import lib.config.config_template as config_template
+import config
+
+class ConfigManagement:
+    """
+    Ensure config file contains all items in config_template and that the values are of the correct type.
+    Apply defaults from config_template where values are missing in config.
+    """
+    def __init__(self) -> None:
+        self.log = uLogger("ConfigManagement")
+        self.error_count = 0
+    
+    def configure_error_handling(self) -> None:
+        """
+        Register errors with the error handler for the configuration management module.
+        """
+        from lib.error_handling import ErrorHandler
+        self.error_handler = ErrorHandler("ConfigManagement")
+        self.errors = {
+            "CFG": "Config errors found, check logs"
+        }
+
+        for error_key, error_message in self.errors.items():
+            self.error_handler.register_error(error_key, error_message)
+        
+        return
+        
+    def check_config(self) -> int:
+        self.log.info("Checking config file")
+        
+        for key in dir(config_template):
+            if not hasattr(config, key):
+                self.log.error(f"Missing config item in config.py: {key}, setting default value: {getattr(config_template, key)}")
+                setattr(config, key, getattr(config_template, key))
+                self.error_count += 1
+            else:
+                if type(getattr(config_template, key)) != type(getattr(config, key)) and type(getattr(config_template, key)) != type(None):
```

Comment:
I think `isinstance()` should be used here instread of the type != type.

ALso the `getattr()` will raise an `AttributeError` if you dont pass in a default and the attribute is not found.

Reply:
```python
if not isinstance(getattr(config, key, None), type(getattr(config_template, key, None))) and getattr(config_template, key, None) is not None:
```
would be more pythonic and remove the AttributeError issue.

Reply:
*converts to review comment*

Reply:
Am I missing something? The first IF checks if the config item exists at all, the second IF under the ELSE (item must exist) you've called out here is specifically to identify incorrect types based on the example config type.

Reply:
Ah right I see. 

I still think using `isinstnace` and using `is` to compare `NoneType` values is the pythonic way to go.

if not isinstance(getattr(config, key), type(getattr(config_template, key))) and getattr(config_template, key) is not None:

Reply:
```python
if not isinstance(getattr(config, key), type(getattr(config_template, key))) and getattr(config_template, key) is not None:
    ...
```

(forgot to format it in my previous comment)